distributed-lock-core/src/main/java/com/github/alturkovic/lock/advice/LockMethodInterceptor.java

```diff
@@ -80,7 +81,7 @@ private Object executeLockedMethod(final MethodInvocation invocation, final Lock
   private void scheduleLockRefresh(final LockContext context, final long expiration) {
     final long refresh = intervalConverter.toMillis(context.getLocked().refresh());
     if (refresh > 0) {
-      context.setScheduledFuture(taskScheduler.scheduleAtFixedRate(constructRefreshRunnable(context, expiration), refresh));
+      context.setScheduledFuture(taskScheduler.scheduleAtFixedRate(constructRefreshRunnable(context, expiration), new Date(System.currentTimeMillis() + refresh), refresh));
```

Comment:
I would prefer it to be 
```
if (refresh > 0) {
      Date startTime = Date.from(Instant.now().plusMillis(refresh));
      context.setScheduledFuture(taskScheduler.scheduleAtFixedRate(constructRefreshRunnable(context, expiration), startTime, refresh));
}
```

Also, I believe `com.github.alturkovic.lock.advice.LockBeanPostProcessorTest#shouldRefreshLock` should be changed from 10 or 11 invocations to be 9 or 10 since we are expecting 1 less invocation due to later start.

Reply:
> I would prefer it to be

Updated

> Also, I believe com.github.alturkovic.lock.advice.LockBeanPostProcessorTest#shouldRefreshLock should be changed from 10 or 11 invocations to be 9 or 10 since we are expecting 1 less invocation due to later start.

I ran the tests when I made the changes but I guess I got lucky and never hit 11 invocations.

Updated